things/groups.go

```diff
@@ -0,0 +1,168 @@
+package things
+
+import (
+	"context"
+)
+
+type Group interface {
```

Comment:
I'm also not sure about this idea. Why can't we use Group struct here? This interface resembles struct anyways (explains fields, rather than behavior). It is a limitation compared to the interface, but it also provides better type safety and this looks sufficient for Mainflux needs anyways.

Reply:
Because Thing groups and User groups could differ in implementation and this interface should be common denominator.
Idea was to have at least some parts modular, that I put this into a package and reuse it in Users, Things and Channels groups, no need to rewrite again each endpoint for example.
This is something that was asked for by @drasko, at least one part. I can easily copy paste and refactor this into the usual place

Reply:
@mteodor this is OK, but remove getters and setters, use strucs and change the fields directly via pointer receiver.

Reply:
Group entity itself, however, is the same in both cases, right?